internal/pkg/servicebus/amqpconnection.go

```diff
@@ -180,10 +184,85 @@ func NewAmqpConnection(ctx context.Context, config *types.Configuration) *AmqpCo
 
 	listener.Session = createAmqpSession(&listener)
 	listener.Receiver = createAmqpListener(&listener)
+	listener.ManagementSender, listener.ManagementReceiver, err = listener.createAmqpSBManagementChannels(listener.TopicName, config.ModuleName)
+	if err != nil {
+		log.WithError(err).Error("failed to create management sender, without this renewal of message locks will fail")
+	}
 
 	return &listener
 }
 
+//RenewLocks renews the locks on messages provided
+func (l *AmqpConnection) RenewLocks(ctx context.Context, messages []*amqp.Message) error {
+	lockTokens := make([]amqp.UUID, 0, len(messages))
+	for _, m := range messages {
+		lockTokens = append(lockTokens, m.DeliveryAnnotations["x-opt-lock-token"].(amqp.UUID))
+		log.WithField("uuid", m.DeliveryAnnotations["x-opt-lock-token"].(amqp.UUID)).Info("adding lockid")
+	}
+
+	hostname, _ := os.Hostname()
+	err := l.ManagementSender.Send(ctx, &amqp.Message{
+		ApplicationProperties: map[string]interface{}{
+			"operation": "com.microsoft:renew-lock",
+		},
+		Properties: &amqp.MessageProperties{
+			MessageID: uuid.NewV4().String(),
+			ReplyTo:   hostname + "-receiver",
+		},
+		Value: map[string]interface{}{
+			"lock-tokens": lockTokens,
+		},
+	})
+	if err != nil {
+		log.WithError(err).Error("failed to renew locks on active messages")
+		return err
+	}
+
+	response, err := l.ManagementReceiver.Receive(ctx)
+	if err != nil {
+		log.WithError(err).Error("failed to renew locks on active messages")
```

Comment:
additional message could be inaccurate i.e. we may have successfully renewed the locks but have a subsequent error. maybe better to be "error response from amqp server"

Reply:
Done